### PR TITLE
Enable embark v03 clippy lints

### LINKED ISSUE #159: Update to v0.3 of Embark's standard lints
Embark uses a standard set of lints across all repositories. We just upgraded
to version 0.3 of these lints, and this repository needs to be updated to
the new set of lints with any warnings fixed.

### Steps

1. Copy the contents of the [`lints.rs`] file from
   [EmbarkStudios/rust-ecosystem] and replace the old list of lints in the
   `src/lib.rs` and/or `src/main.rs` files with the new list.
2. Run `cargo clippy`
3. Update and fix any code that now triggers a new warning.

[`lints.rs`]: https://github.com/EmbarkStudios/rust-ecosystem/blob/main/lints.rs
[EmbarkStudios/rust-ecosystem]: https://github.com/EmbarkStudios/rust-ecosystem/

### PR DESCRIPTION
**Pull request by celialewis3**
Fixed #159 to update clippy lints in `lib.rs` to Embark's v03 clippy [lints](https://github.com/EmbarkStudios/rust-ecosystem/blob/main/lints.rs).

### CODE REVIEW

**Comment by Jake-Shadle** [comment]
You'll need to add the lints to `main.rs` as well since this is both a library and a binary.

File: lib/src/lib.rs [hunk 1]
```diff
@@ -1,19 +1,49 @@
...
+    clippy::suboptimal_flops,
+    clippy::todo,
+    clippy::unimplemented,
+    clippy::unnested_or_patterns,
+    clippy::unused_self,
+    clippy::verbose_file_reads,
     future_incompatible,
-    nonstandard_style
+    nonstandard_style,
+    rust_2018_idioms
```
Comment (repi) [inline lib/src/lib.rs:46]: Make sure to copy the entire file contents of https://github.com/EmbarkStudios/rust-ecosystem/blob/main/lints.rs for the lints so one gets the `/// BEGIN` / `/// END` lines as well so one sees the version and source of it. and is easier to upgrade later

**Review by repi — CHANGES_REQUESTED** [review]

File: lib/src/lib.rs (hunk 1 above)
Comment (celialewis3) [reply lib/src/lib.rs:46]: Will do

**Review by repi — APPROVED** [review]

**Comment by celialewis3** [comment]
I added the lints to `main.rs` as well as copied the full contents from `BEGIN` to `END`. However I had to leave out `#![deny(unsafe_code)]` in `lib.rs` because we have several unsafe blocks in there.

**Comment by Jake-Shadle** [comment]
That's fine, the lints are generally applicable, but don't fit for all of our current code.

**Comment by repi** [comment]
> 
> 
> I added the lints to `main.rs` as well as copied the full contents from `BEGIN` to `END`. However I had to leave out `#![deny(unsafe_code)]` in `lib.rs` because we have several unsafe blocks in there.

Don't leave it out in the template, do an `allow(unsafe_code)` after it instead. the template should be unchanged but it is fine to override after it in the crate-specific section. can look a bit silly but it is so that we can just copy'n'paste the template to any repo unmodified

File: lib/src/lib.rs [hunk 2]
```diff
@@ -48,6 +48,9 @@
     rust_2018_idioms
 )]
 // END - Embark standard lints v0.3
+// crate-specific exceptions:
+#![allow(unsafe_code)]
+#![allow(clippy::exit)]
```
Comment (Jake-Shadle) [inline lib/src/lib.rs:53]:
```suggestion

// crate-specific exceptions:
#![allow(unsafe_code, clippy::exit)]
```

**Review by Jake-Shadle — CHANGES_REQUESTED** [review]

**Review by Jake-Shadle — APPROVED** [review]

**Pull request merged**